arviz/plots/backends/bokeh/pairplot.py

```diff
@@ -42,156 +49,123 @@ def plot_pair(
+            source=source, filters=[GroupFilter(column_name=divergenve_name, group="0")]
+        )
+        source_div = CDSView(
+            source=source, filters=[GroupFilter(column_name=divergenve_name, group="1")]
+        )
 
-        tmp_flat_var_names = None
-        if len(flat_var_names) == len(list(set(flat_var_names))):
-            source_dict = dict(zip(flat_var_names, [list(post) for post in infdata_group]))
-        else:
-            tmp_flat_var_names = ["{}__{}".format(name, str(uuid4())) for name in flat_var_names]
-            source_dict = dict(zip(tmp_flat_var_names, [list(post) for post in infdata_group]))
-        if divergences:
-            divergenve_name = "divergences_{}".format(str(uuid4()))
-            source_dict[divergenve_name] = (
-                np.array(diverging_mask).astype(bool).astype(int).astype(str)
-            )
+    # pylint: disable=R1702
+    for i in range(0, numvars - var):
 
-        source = ColumnDataSource(data=source_dict)
+        var1 = flat_var_names[i] if tmp_flat_var_names is None else tmp_flat_var_names[i]
 
-        if divergences:
-            source_nondiv = CDSView(
-                source=source, filters=[GroupFilter(column_name=divergenve_name, group="0")]
-            )
-            source_div = CDSView(
-                source=source, filters=[GroupFilter(column_name=divergenve_name, group="1")]
-            )
+        for j in range(0, numvars - var):
 
-        for i in range(0, numvars - 1):
-            var1 = flat_var_names[i] if tmp_flat_var_names is None else tmp_flat_var_names[i]
-
-            for j in range(0, numvars - 1):
-                if j < i:
-                    continue
+            var2 = (
+                flat_var_names[j + var]
+                if tmp_flat_var_names is None
+                else tmp_flat_var_names[j + var]
+            )
 
-                var2 = (
-                    flat_var_names[j + 1]
-                    if tmp_flat_var_names is None
-                    else tmp_flat_var_names[j + 1]
+            if j == i and diagonal:
+                rotate = numvars == 2 and j == 1
+                var1_dist = infdata_group[i]
+                plot_dist(
+                    var1_dist,
+                    ax=ax[j, i],
+                    show=False,
+                    backend="bokeh",
+                    rotated=rotate,
+                    **marginal_kwargs,
                 )
 
-                if kind == "scatter":
+            elif j + var > i:
+
+                if "scatter" in kind:
                     if divergences:
                         ax[j, i].circle(var1, var2, source=source, view=source_nondiv)
                     else:
                         ax[j, i].circle(var1, var2, source=source)
 
-                elif kind == "kde":
+                if "kde" in kind:
                     var1_kde = infdata_group[i]
-                    var2_kde = infdata_group[j + 1]
+                    var2_kde = infdata_group[j + var]
```

Comment:
we can define var1_ary and var2_ary outside the ifs and use them when needed, kde, hexbin, point_estimates...